File: VB/FunctionBindingExample/Common/DataItem.vb
Imports System
Imports System.Linq
Imports System.Collections.Generic
Imports DevExpress.Mvvm.POCO

Namespace FunctionBindingExample.Common
    Public Class DataItem
        Protected Sub New(ByVal [date] As Date, ByVal valueSummary As Double, ByVal stageName As String)
            Me.Date = [date]
            Me.ValueSummary = valueSummary
            Me.StageName = stageName
        End Sub

        Public Shared Function Create(ByVal [date] As Date, ByVal valueSummary As Double, ByVal stageName As String) As DataItem
            Return ViewModelSource.Create(Function() New DataItem([date], valueSummary, stageName))
        End Function

        Public Property [Date]() As Date
        Public Property ValueSummary() As Double
        Public Property StageName() As String
    End Class
End Namespace

File: CS/FunctionBindingExample/Common/DataItem.cs
using DevExpress.Mvvm.POCO;
using System;

namespace FunctionBindingExample.Common {
    public class DataItem {
        protected DataItem(DateTime date, double valueSummary, string stageName) {
            Date = date;
            ValueSummary = valueSummary;
            StageName = stageName;
        }

        public static DataItem Create(DateTime date, double valueSummary, string stageName) {
            return ViewModelSource.Create(() => new DataItem(date, valueSummary, stageName));
        }

        public DateTime Date { get; set; }
        public double ValueSummary { get; set; }
        public string StageName { get; set; }
    }
}
